### ISSUE TITLE
Stack overflow when Mesh tolerance is 0

### ISSUE THREAD

**Issue by spacekookie** [ggez/ggez#892]
**Describe the bug**
When trying to construct a mesh with tolerance 0.0, the binary stack overflows (and aborts)

**To Reproduce**

```
// Inside EventHandler trait
    fn draw(&mut self, ctx: &mut Context) -> GameResult<()> {
        graphics::clear(ctx, graphics::Color::from_rgb(15, 15, 15));

        let circ = Mesh::new_circle(
            ctx,
            DrawMode::fill(),
            Point2::from([0.0, 0.0]),
            35.0,
            0.0,
            graphics::WHITE,
        )?;

        circ.draw(ctx, DrawParam::new())?;

        graphics::present(ctx)
    }
```

**Expected behavior**
A mesh is rendered.

**Hardware and Software:**
 - ggez version: 0.6.0-rc0
 - OS: Linux (NixOS)
 - Graphics card: AMD RX Vega 64
 - Graphics card drivers: Linux mainline AMDGPU

**Comment by Ratysz** [comment]
This is a [`lyon`](https://github.com/nical/lyon) issue, not a `ggez` issue. I would also argue that this is not even an issue, see [docs](https://docs.rs/lyon_geom/0.11.0/lyon_geom/#flattening); tolerance of 0 practically means "draw me a perfect circle out of straight line segments", which would require an infinite number of segments.

**Comment by spacekookie** [comment]
Can you describe this in the actual ggez docs then too?
Because as a casual user of this library I wasn't aware that it even 
used lyon under the hood.

**Comment by Ratysz** [comment]
That would be best, yes; to be fair, [some](https://docs.rs/ggez/0.5.1/ggez/graphics/struct.MeshBuilder.html#method.circle) docs already do, just not [all](https://docs.rs/ggez/0.5.1/ggez/graphics/struct.Mesh.html#method.new_circle) that should.

**Comment by icefoxen** [comment]
Stack overflow is a kinda nasty result though, we can at least check for that and give a nicer error.

**Comment by icefoxen** [comment]
Resolved in 39c02bc, added an assert that gives a nicer error message when `tolerance` is 0.  Thanks for the bug report!  :heart:

**Comment by spacekookie** [comment]
Thanks for making ggez :)
